liquidprompt

```diff
@@ -3418,6 +3432,126 @@ _lp_temperature_color() {
     lp_temperature_color="${LP_MARK_TEMP}${ret}${lp_temperature}°${NO_COL}"
 }
 
+######################
+# Current Disk Usage #
+######################
+
+# Convert bytes in human-readable, decimal-prefixed notation.
+# TODO: use IEC binary prefixes (using powers of two) instead of decimal ones?
+# NOTE: lp_human returns a space-separated string with value and then unit,
+#       so that the caller may split value from unit and display them differently.
+__lp_bytes_to_human() { # bytes [precision]
```

Comment:
Oof, yeah this function is rough. I haven't speed tested this, but I bet it's not great. A few things I see at first glance:
* `local units=` can be pulled out of the function to save some time.
* We can very much use integer maths, might need to do some bit shifting or `* 1000` or something. I thought I was doing that already somewhere in the project.

Reply:
Here is a better function I found:
https://unix.stackexchange.com/a/259254/421569
```bash
# Converts bytes value to human-readable string [$1: bytes value]
bytesToHumanReadable() {
    local i=${1:-0} d="" s=0 S=("Bytes" "KiB" "MiB" "GiB" "TiB" "PiB" "EiB" "YiB" "ZiB")
    while ((i > 1024 && s < ${#S[@]}-1)); do
        printf -v d ".%02d" $((i % 1024 * 100 / 1024))
        i=$((i / 1024))
        s=$((s + 1))
    done
    echo "$i$d ${S[$s]}"
}
```
It clearly has a few issues (readability for sure), mainly that `printf -v <var>` wasn't supported in Zsh until 5.3.

Here is my suggestion:
```bash
__lp_bytes_to_human() { # bytes
    local i="$1" size remainder=""
    # This has to be split on multiple lines for Zsh 5.0 (see wiki)
    local units=()
    # Not much point in going past PiB, as Bash and Zsh start failing int math.
    units=("Bytes" "KiB" "MiB" "GiB" "TiB" "PiB" "EiB" "YiB" "ZiB")

    for ((size=0; i > 1024 && size < ${#units[@]}-1; size++)); do
        remainder=$((i % 1024 * 100 / 1024))
        i=$((i / 1024))
    done

    if [[ -n $remainder ]]; then
      ((remainder < 10)) && remainder="0$remainder"
      remainder=".$remainder"
    fi

    lp_bytes="$i$remainder"
    lp_bytes_units="${units[$size]}"
}
```